### ISSUE TITLE
Allowing to split series/frames into chunks within Python

### ISSUE THREAD

**Issue by aldanor** [pola-rs/polars#11550]
### Description

Given the current ffi behaviour (aka "rechunk everything into Array"), there's one part missing from Python API that would otherwise allow for some workarounds if you need some control over rechunking (e.g. sending frames over to pyo3-polars without rechunking).

Would something like this be possible?
```py
# Series
def split_chunks(self) -> list[Series]: ...

# DataFrame
def split_chunks(self) -> list[DataFrame]: ...
```

For empty or single-chunk, return the object itself; otherwise, construct contiguous series/frames from the underlying chunks.

**Comment by aldanor** [comment]
Another method to add might be `chunk_sizes()`.

**Comment by ritchie46** [comment]
We already got `Series.get_chunks` which does this for `Series`. We could add the same for `DataFrame`, however that might still cause a realloc as a `DataFrame` doesn't guarantee all chunks are aligned.

**Comment by aldanor** [comment]
> We already got Series.get_chunks which does this for Series. We could add the same for DataFrame, however that might still cause a realloc as a DataFrame doesn't guarantee all chunks are aligned.

Right, I didn't actually notice Series already have them since I've been looking at frames, that's great!

For frames... 
- Add an optional parameter `DataFrame.get_chunks(align_chunks=True/False)`?
- Add `.align_chunks()` method?

All this plus `chunk_sizes()` will basically expose most of the 'chunking arsenal' which exists on the Rust side but is hidden away from Python.
